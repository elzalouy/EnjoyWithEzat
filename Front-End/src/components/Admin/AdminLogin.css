input{
  border-width: 0px 0px 1px 0px !important;
  border-radius: 0 !important;
    background-color: transparent !important;
    box-shadow: transparent;
}
.Page{
  padding: 0;
  margin: 0;
  height: 100% !important;
  width: 100% !important;
  overflow: hidden;
  color: white;
  background-color: gold ;
}
.Form{
  border-radius: 5px;
  margin-top: 50px;
  text-align: center;
  background-color: #1e1e1e !important;
  transition: all 0.5s;
  overflow: hidden;
}
.Form-Submit{
  border-radius: 0px !important;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  background-color:#252526 !important;
  color: white !important;
  width: 100%;
}
.Form-Submit:hover{
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}
