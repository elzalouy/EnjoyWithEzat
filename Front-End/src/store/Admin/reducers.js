export const onHandleChange=(state,action)=>{
    state[action.payload.element]=action.payload.value;
}
export const onLoginError=(state,action)=>{
    state.onLoginError=action.payload;
}
const exports={onHandleChange,onLoginError};
export default exports;