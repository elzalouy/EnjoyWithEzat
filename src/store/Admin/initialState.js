const initialState={
    name:'ezat elzalouy',
    email:'',
    password:'',
    role:'all',
    onLoginError:''
}
export default initialState;