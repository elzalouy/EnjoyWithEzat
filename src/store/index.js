import { configureStore } from "@reduxjs/toolkit";
import UiSlice from "./Ui/index";
import PersonSlice from "./Person/index";
import Projects from './Projects/index'
const store = configureStore({
  reducer: { UI: UiSlice, Person: PersonSlice, Projects: Projects},
});
export default store;
