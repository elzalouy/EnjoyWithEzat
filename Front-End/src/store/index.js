import { configureStore } from "@reduxjs/toolkit";
import UiSlice from "./Ui/index";
import PersonSlice from "./Profile/index";
import Projects from './Projects/index'
import Admin from "./Admin/index";
const store = configureStore({
  reducer: { UI: UiSlice, Person: PersonSlice, Projects: Projects, Admin : Admin },
});
export default store;