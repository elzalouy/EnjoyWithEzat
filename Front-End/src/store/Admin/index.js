import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";
import reducers from './reducers';
const AdminSlice = createSlice({
  name: "admin",
  initialState,
  reducers: reducers
});
export const AdminActions=AdminSlice.actions;
export default AdminSlice.reducer;
