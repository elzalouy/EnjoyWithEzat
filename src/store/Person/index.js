import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";

const PersonSlice = createSlice({
  name: "person",
  initialState,
  reducers: {
    selectProjectsByType(state,action){
    }
  },
});
export const PersonActions = PersonSlice.actions;
export default PersonSlice.reducer;
