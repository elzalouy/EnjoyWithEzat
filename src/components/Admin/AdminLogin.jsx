import { useDispatch, useSelector } from "react-redux";
import { validate } from "../../httpServices/admin";
import { AdminActions } from "../../store/Admin";
import Input from "../UIs/Input/Input";
import "./AdminLogin.css";

const AdminLogin = () => {
  const dispatch = useDispatch();
  const admin = useSelector((state) => state.Admin);
  
    const handleSubmit = (e) => {
      e.preventDefault();
      alert("Working on it now!");
    };
    const handleChange = (e) => {
      const result = validate(admin);
      if (result.message) dispatch(AdminActions.onLoginError(result.message));
      else dispatch(AdminActions.onLoginError(""));
      dispatch(
        AdminActions.onHandleChange({
          element: e.target.name,
          value: e.target.value,
        })
      );
    };
  
  const inputs = [
    {
      label: { for: "email", icon: "fas fa-at" },
      input: {
        type: "email",
        id: "email",
        name: "email",
        value: admin.email,
        handleChange: handleChange,
      },
    },  
    {
      label: { for: "password", icon: "fas fa-lock" },
      input: {
        type: "password",
        id: "password",
        name: "password",
        value: admin.password,
        handleChange: handleChange,
      },  
    },  
  ];

  return (
      <div className="Page" name='page'>
        <div className="row m-4">
          <div className="col-sm-4"></div>
          <div className="Form col-md-4 Shadow-1">
            <i className="fas fa-user-tie w pv-2"></i>
            <form
              onSubmit={handleSubmit}
              className="row pt-5"
              name="adminloginform"
            >
              <Input {...inputs[0]} />
              <Input {...inputs[1]} />
              <button type="submit" className="btn Form-Submit">
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
  );
};

export default AdminLogin;