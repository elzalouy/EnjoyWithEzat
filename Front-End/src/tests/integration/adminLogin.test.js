/**
 * 1- React-test-renderer
 * This package provides a React renderer that can be used to render React components to pure JavaScript objects, 
 * without depending on the DOM or a native mobile environment.
 * 2- Snapshot
 * Basically a Snapshot Test creates a snapshot 
 * -- which is stored in a separate file -- of your rendered component's output when you run your test.
 *  This snapshot is used for diffing it to the next snapshot when you run your test again.
 *  If your rendered component's output has changed, the diff of both snapshots will show it and the Snapshot Test will fail.
 */

import React from 'react';
import { Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import createMockStore from 'redux-mock-store';
import AdminLogin from '../../components/Admin/AdminLogin';
import { AdminActions } from '../../store/Admin';

const mockstore=createMockStore([]);
describe('Admin login component',()=>{
    let store,component;
        store=mockstore({Admin:{email:'',password:"",role:"",name:"ezat",onLoginError:''}});
        store.dispatch=jest.fn();
        component=renderer.create(<Provider store={store}><AdminLogin/></Provider>);  
    it('it should render with given state from redux store',()=>{
        expect(component.toJSON()).toMatchSnapshot();
    });
    it('it should call dispatch',()=>{
         renderer.act(()=>{
            component.root.findByProps({name:'email',id:'email',type:'email'}).props.onChange({target:{value:"eza",name:"email"}});
        })
        expect(store.dispatch).toHaveBeenCalledTimes(2);
        expect(store.dispatch).toHaveBeenNthCalledWith(1,AdminActions.onLoginError("\"email\" is not allowed to be empty"));
        expect(store.dispatch).toHaveBeenLastCalledWith(AdminActions.onHandleChange({element:"email",value:"eza"}));
    });
});


