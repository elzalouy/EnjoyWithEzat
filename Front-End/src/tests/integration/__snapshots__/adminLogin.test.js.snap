// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin login component it should render with given state from redux store 1`] = `
<div
  className="Page"
  name="page"
>
  <div
    className="row m-4"
  >
    <div
      className="col-sm-4"
    />
    <div
      className="Form col-md-4 Shadow-1"
    >
      <i
        className="fas fa-user-tie large-5 pv-2"
      />
      <form
        className="row pt-5"
        name="adminloginform"
        onSubmit={[Function]}
      >
        <div
          className="col-12 row pv-2 text-left"
        >
          <label
            className="form-label col-2 pt-2"
            htmlFor="email"
          >
            <i
              className="fas fa-at"
            />
          </label>
          <input
            className="form-control col-10"
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
        </div>
        <div
          className="col-12 row pv-2 text-left"
        >
          <label
            className="form-label col-2 pt-2"
            htmlFor="password"
          >
            <i
              className="fas fa-lock"
            />
          </label>
          <input
            className="form-control col-10"
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
        <button
          className="btn Form-Submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
